### PR TITLE
APEXMALHAR-2136 1) Fixed the null pointer exception issue. 2) Added getter and setter for IncrementalCheckpointManager

### PR DESCRIPTION
**Pull request by chaithu14**

### CODE REVIEW

**Comment by chaithu14** [comment]
@chandnisingh Please review and merge.

File: library/src/main/java/org/apache/apex/malhar/lib/state/managed/AbstractManagedStateImpl.java [hunk 1]
```diff
@@ -203,7 +203,12 @@ public void setup(OperatorContext context)
       //delete all the wal files with windows > activationWindow.
       //All the wal files with windows <= activationWindow are loaded and kept separately as recovered data.
       try {
-        for (long recoveredWindow : checkpointManager.getWindowIds(operatorContext.getId())) {
+        long[] recoveredWindows = checkpointManager.getWindowIds(operatorContext.getId());
+        if (recoveredWindows == null) {
+          readerService = Executors.newFixedThreadPool(numReaders, new NameableThreadFactory("managedStateReaders"));
```
Comment (otterc) [inline library/src/main/java/org/apache/apex/malhar/lib/state/managed/AbstractManagedStateImpl.java]: This is already being done at the end. why do you need to duplicate the stmt here?

File: library/src/main/java/org/apache/apex/malhar/lib/state/managed/AbstractManagedStateImpl.java [hunk 2]
```diff
@@ -203,7 +203,12 @@ public void setup(OperatorContext context)
...
+          return;
+        }
+        for (long recoveredWindow : recoveredWindows) {
```
Comment (otterc) [inline library/src/main/java/org/apache/apex/malhar/lib/state/managed/AbstractManagedStateImpl.java]: You can put this block within `if (recoveredWindows == null)`

Comment (chaithu14) [inline library/src/main/java/org/apache/apex/malhar/lib/state/managed/AbstractManagedStateImpl.java]: Done

File: library/src/main/java/org/apache/apex/malhar/lib/state/managed/AbstractManagedStateImpl.java [hunk 3]
```diff
@@ -536,6 +538,16 @@ public void setDurationPreventingFreeingSpace(Duration durationPreventingFreeing
     this.durationPreventingFreeingSpace = durationPreventingFreeingSpace;
   }
 
+  public IncrementalCheckpointManager getCheckpointManager()
+  {
+    return checkpointManager;
+  }
+
+  public void setCheckpointManager(IncrementalCheckpointManager checkpointManager)
```
Comment (otterc) [inline library/src/main/java/org/apache/apex/malhar/lib/state/managed/AbstractManagedStateImpl.java]: `@NotNull` to checkpointManager and please use Preconditions.checkNotNull(...)

Comment (chaithu14) [inline library/src/main/java/org/apache/apex/malhar/lib/state/managed/AbstractManagedStateImpl.java]: Updated as per the comment.

**Pull request merged**